src/ol/format/wmsgetfeatureinfoformat.js

```diff
@@ -86,7 +97,13 @@ ol.format.WMSGetFeatureInfo.prototype.readFeatures_ = function(node, objectStack
           'localName of layer node should match layerIdentifier');
 
       var toRemove = ol.format.WMSGetFeatureInfo.layerIdentifier_;
-      var featureType = layer.localName.replace(toRemove, '') +
+      var layerName = layer.localName.replace(toRemove, '');
+
+      if (this.layers_ && ol.array.includes(this.layers_, layerName)) {
```

Comment:
The code was `this.layers_.indexOf(layerName) === -1` so it should be `!ol.array.includes(this.layers_, layerName)` ?